static/app/views/performance/landing/widgets/widgets/lineChartListWidget.tsx

```diff
@@ -56,98 +58,126 @@ export function LineChartListWidget(props: Props) {
     );
   }
 
-  const Queries = {
-    list: useMemo<QueryDefinition<DataType, WidgetDataResult>>(
-      () => ({
-        fields: props.fields[0],
-        component: provided => {
-          const eventView = provided.eventView.clone();
-          eventView.sorts = [{kind: 'desc', field: props.fields[0]}];
-          if (props.chartSetting === PerformanceWidgetSetting.MOST_RELATED_ISSUES) {
-            eventView.fields = [
-              {field: 'issue'},
-              {field: 'transaction'},
-              {field: 'title'},
-              {field: 'project.id'},
-              {field: props.fields[0]},
-            ];
-            eventView.additionalConditions.setFilterValues('event.type', ['error']);
-            eventView.additionalConditions.setFilterValues('!tags[transaction]', ['']);
-            const mutableSearch = new MutableSearch(eventView.query);
-            mutableSearch.removeFilter('transaction.duration');
-            eventView.query = mutableSearch.formatString();
-          } else {
-            eventView.fields = [
-              {field: 'transaction'},
-              {field: 'project.id'},
-              {field: props.fields[0]},
-            ];
-          }
-          return <DiscoverQuery {...provided} eventView={eventView} limit={3} />;
-        },
-        transform: transformDiscoverToList,
-      }),
-      [props.eventView, props.fields, props.chartSetting, props.organization.slug]
-    ),
-    chart: useMemo<QueryDefinition<DataType, WidgetDataResult>>(
-      () => ({
-        enabled: widgetData => {
-          return !!widgetData?.list?.data?.length;
-        },
-        fields: props.fields[0],
-        component: provided => {
-          const eventView = provided.eventView.clone();
-          eventView.additionalConditions.setFilterValues('transaction', [
-            provided.widgetData.list.data[selectedListIndex].transaction as string,
+  const slowList = [
+    PerformanceWidgetSetting.SLOW_HTTP_OPS,
+    PerformanceWidgetSetting.SLOW_DB_OPS,
+    PerformanceWidgetSetting.SLOW_BROWSER_OPS,
+    PerformanceWidgetSetting.SLOW_RESOURCE_OPS,
+    PerformanceWidgetSetting.MOST_SLOW_FRAMES,
+    PerformanceWidgetSetting.MOST_FROZEN_FRAMES,
+  ];
+  const isSlowestType = slowList.includes(props.chartSetting);
```

Comment:
Nit: Can we declare `slowList` as a set outside of this function

Reply:
Yeah, I'm also considering splitting the file. I'm trying to think if I can simplify this memoization code or list visualizations to reduce the boilerplate between widgets first.